### PR TITLE
War Financing Rework

### PR DESCRIPTION
**Pull request by Cernelius**

### CODE REVIEW

File: map/games/Civil_War.xml [hunk 1]
```diff
@@ -14552,45 +14571,50 @@
...
       <option name="notification" value="ConfederateDefault"/>
+      <option name="players" value="Confederate:Union"/>
+    </attachment>
+    <attachment name="triggerAttachment_Confederate_Default_Units" attachTo="Confederate" javaClass="games.strategy.triplea.attachments.TriggerAttachment" type="player">
+      <option name="conditions" value="conditionAttachment_Confederate_HasBackPayment_Anywhere"/>
+      <option name="when" value="after:ConfederateUserActions"/>
+      <option name="removeUnits" value="all:ConfederatePayment:ConfederateDebt:infantry:elite_infantry:cavalry:elite_cavalry:artillery:general:gunboat:ironclad:frigate:battleship:submarine:engineers" count="10000"/>
       <option name="players" value="Confederate"/>
       <option name="resource" value="PUs"/>
       <option name="resourceCount" value="-10000"/>
```
Comment (ron-murhammer) [inline map/games/Civil_War.xml]: @Cernelius Why does this trigger and the one below both remove all PUs?

Comment (Cernelius) [reply map/games/Civil_War.xml]: I've not actually read and checked the whole original code, just focused on what was clearly wrong or I wanted it working differently, so here you have spotted a case of redundant coding I've overlooked.

This is the original code:
```
    <attachment name="triggerAttachment_Confederate_Default" attachTo="Confederate" javaClass="games.strategy.triplea.attachments.TriggerAttachment" type="player">
      <option name="conditions" value="conditionAttachment_HasBackPayment_Anywhere"/>
      <option name="when" value="after:ConfederateUserActions"/>
      <option name="removeUnits" value="all:ConfederatePayment:ConfederateDebt:infantry:elite_infantry:cavalry:elite_cavalry:artillery:general:gunboat:ironclad:frigate:battleship:submarine:engineers" count="600"/>
      <option name="notification" value="ConfederateDefault"/>
      <option name="players" value="Confederate"/>
      <option name="resource" value="PUs"/>
      <option name="resourceCount" value="-10000"/>
    </attachment>
    <attachment name="triggerAttachment_Confederate_Default_PUs" attachTo="Confederate" javaClass="games.strategy.triplea.attachments.TriggerAttachment" type="player">
      <option name="conditions" value="conditionAttachment_HasBackPayment_Anywhere"/>
      <option name="when" value="after:ConfederateUserActions"/>
      <option name="players" value="Confederate"/>
      <option name="resource" value="PUs"/>
      <option name="resourceCount" value="-10000"/>
    </attachment>
```
Same thing for the Union.

So, yes, well spotted: the last two options of "triggerAttachment_Confederate_Default_Units" and "triggerAttachment_Union_Default_Units" should be removed.

I wouldn't be surprised there are more redundant things around.

I'll push the fix.

Comment (Cernelius) [reply map/games/Civil_War.xml]: The changes I made on that (beside renaming) are:
- Take out the notification part, to another trigger, so to display it to both players (not only the defaulting one).
- Change the removal unit count from 600 to 10000 (doesn't matter, but 10000 usually means "all").

Didn't notice the next trigger was purely redundant.

Comment (Cernelius) [reply map/games/Civil_War.xml]: Also noticed now that I've not specified the "u" units in the Union one. So that is an actual xml bug in my coding.

Comment (Cernelius) [reply map/games/Civil_War.xml]: Nevermind. I just checked: none of the units removed are the "u" ending ones; so all good there.

Comment (Cernelius) [reply map/games/Civil_War.xml]: The second commit I just made should address that redundancy you pointed out (I've imported from the original code). I wouldn't be surprised if there are more cases.

**Comment by Cernelius** [comment]
Talking about the default triggers, I noticed the descriptions in notes and notifications for the units removal due to default are all wrong; so this third commit should address that.

**Pull request merged**